Cubical/Algebra/Group/Abelianization/AbelianizationAsCoeq.agda

```diff
@@ -0,0 +1,398 @@
+  renaming (Abelianization to HITAbelianization;
+            η to HITη;
+            comm to HITcomm;
+            isset to HITisset)
+
+open import Cubical.Algebra.Group.Abelianization.Properties
+  hiding (elimProp; elimProp2; elimProp3; elimContr; elimContr2; rec; rec2)
+  renaming (AbelianizationAbGroup to HITasAbelianGroup;
+            AbelianizationHom to HITηAsGroupHom)
+open Cubical.Algebra.Group.Abelianization.Properties.UniversalProperty
+  renaming (inducedHom to HITinducedHom;
+            commutativity to HITcommutativity;
+            uniqueness to HITuniqueness)
+
+private
+  variable
+    ℓ : Level
+    G : Group ℓ
+
+module AbelianizationAsCoeq (G : Group ℓ) where
+  open GroupStr {{...}}
+  open GroupTheory G
+  private
+    instance
+      _ = snd G
+    G' = fst G
+
+  {-
+    The definition of the abelianization of a group as a coequalizer of sets. The generality
+    of the coequalized functions will be needed to define the group structure on the abelianization.
+  -}
+  Abelianization : Type ℓ
+  Abelianization =
+    SetCoequalizer {A = G' × G' × G'} {B = G'}
+      (λ (x , y , z) → x · y · z)
+      (λ (x , y , z) → x · z · y)
+
+  -- some convenient relabellings, so it looks more similar to the abelianization as a HIT.
+  incAb : G' → Abelianization
+  incAb = inc
+
+  comm : (x y z : G') → incAb (x · (y · z)) ≡ incAb (x · (z · y))
+  comm = λ x y z → coeq (x , y , z)
+
+  -- Definition of the group structure on the abelianization. Here the generality of the coequalized functions is used.
+  _·Ab_ : Abelianization → Abelianization → Abelianization
+  _·Ab_ =
+    rec2
+      squash
+      (λ x y → incAb (x · y))
+      (λ (a , b , c) d → incAb ((a · (b · c)) · d) ≡⟨ cong (λ x → incAb (x · d)) (assoc _ _ _) ⟩
+                         incAb (((a · b) · c) · d) ≡⟨ cong incAb (sym (assoc (a · b) c d)) ⟩
+                         incAb ((a · b) · (c · d)) ≡⟨ comm (a · b) c d ⟩
+                         incAb ((a · b) · (d · c)) ≡⟨ cong incAb (sym (assoc _ _ _)) ⟩
+                         incAb (a · (b · (d · c))) ≡⟨ cong (λ x → incAb (a · x)) (assoc _ _ _) ⟩
+                         incAb (a · ((b · d) · c)) ≡⟨ comm a (b · d) c ⟩
+                         incAb (a · (c · (b · d))) ≡⟨ cong (λ x → incAb (a · x)) (assoc _ _ _) ⟩
+                         incAb (a · ((c · b) · d)) ≡⟨ cong incAb (assoc a (c · b) d) ⟩
+                         incAb ((a · (c · b)) · d) ∎)
+      (λ (b , c , d) a → incAb (a · (b · (c · d))) ≡⟨ cong incAb (assoc _ _ _) ⟩
+                         incAb ((a · b) · (c · d)) ≡⟨ comm (a · b) c d ⟩
+                         incAb ((a · b) · (d · c)) ≡⟨ cong incAb (sym (assoc _ _ _)) ⟩
+                         incAb (a · (b · (d · c))) ∎)
+
+  1Ab : Abelianization
+  1Ab = incAb 1g
+
+  invAb : Abelianization → Abelianization
+  invAb =
+    rec
+      squash
+      (λ x → incAb (inv x))
+      (λ (a , b , c) → incAb (inv (a · (b · c)))              ≡⟨ cong incAb (invDistr a (b · c)) ⟩
```

Comment:
lines could easily be shortened, by line breaking before the arrow...

Reply:
done